### ISSUE TITLE
X.H.WallpaperSetter with PNG support?

### ISSUE THREAD

**Issue by oogeek** [xmonad/xmonad-contrib#518]
### Problem Description
I am wondering why as stated in `X.H.WallpaperSetter`, the background image should be JPG format.
I have read the source code and can not figure it out. 

Specifically, I don't understand why jpg is used here.
```
-- |returns the default association list (maps name to name.jpg, non-alphanumeric characters are omitted)
defWPNames :: [WorkspaceId] -> WallpaperList
defWPNames xs = WallpaperList $ map (\x -> (x,WallpaperFix (filter isAlphaNum x++".jpg"))) xs
```
We can actually take the last few characters and use it as suffix. Then we can have `abc.png` or `abc.jpg`

Another question is, is it necessary to `convert` the image now:
```
-- | Gets a list of geometry rectangles and filenames, builds and sets wallpaper
applyWallpaper :: [(Rectangle, FilePath)] -> X ProcessHandle
applyWallpaper parts = do
  winset <- gets windowset
  let (vx,vy) = getVScreenDim winset
  layers <- liftIO $ mapM layerCommand parts
  let basepart ="convert -size "++show vx++"x"++show vy++" xc:black "
      endpart =" jpg:- | feh --no-xinerama --bg-tile --no-fehbg -"
      cmd = basepart ++ (concat $ intersperse " " layers) ++ endpart
  liftIO $ runCommand cmd

```
I believe that `feh` can handle that if other options are used. And it is unnecessary if the background picture is just the same size as the screen.
### Checklist

  - [X] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [ ] I tested my configuration with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

**Comment by TheMC47** [comment]
I think you're making valid points, but there's always the concern of staying backward-compatible (not that I think a lot of people are using the module).
I say we break compatibility because it feels like a net improvement.

**Comment by slotThe** [comment]
On Fri, Apr 16 2021 16:42, Yecine Megdiche wrote:
> I think you're making valid points, but there's always the concern of
> staying backward-compatible (not that I think a lot of people are
> using the module).
> I say we break compatibility because it feels like a net improvement. 

I feel like the .jpg -> allow more img formats direction should be able
to stay backwards compatible.  

I admit that I haven't looked over the code very much, but I'm not
certain that the second point holds and feh can do everything that
convert is doing right now, especially considering the modules claim:

> This was especially tested with multi-head setups - if you have two
> monitors and swap the workspaces, the wallpapers will be swapped too,
> scaled accordingly and rotated if necessary (e.g. if you are using
> your monitor rotated but only have wide wallpapers).

**Comment by oogeek** [comment]
> This was especially tested with multi-head setups - if you have two monitors and swap the workspaces, the wallpapers will be swapped too, scaled accordingly and rotated if necessary (e.g. if you are using your monitor rotated but only have wide wallpapers).
Yeah, so I agree with that. So we can change the code to support for more photo format but leave the second point alone.

**Comment by slotThe** [comment]
On Fri, Apr 16 2021 23:21, oogeek wrote:
> Yeah, so I agree with that. So we can change the code to support for
> more photo format but leave the second point alone.

Sounds good to me!

**Comment by oogeek** [comment]
> On Fri, Apr 16 2021 23:21, oogeek wrote: Yeah, so I agree with that. So we can change the code to support for more photo format but leave the second point alone.
> Sounds good to me!

I think it is relatively easy to fix the first point. Let me open a PR later.

**Comment by slotThe** [comment]
Closed by #521
